code/modules/admin/topic.dm

```diff
@@ -349,9 +349,15 @@
 		if(!check_rights(R_SERVER))
 			return
 
+		if(!SSticker.delay_end)
+			SSticker.admin_delay_notice = input(usr, "Enter a reason for delaying the round end", "Round Delay Reason") as null|text
+			if(SSticker.admin_delay_notice == null)
+				return
 		SSticker.delay_end = !SSticker.delay_end
-		log_admin("[key_name(usr)] [SSticker.delay_end ? "delayed the round end" : "has made the round end normally"].")
-		message_admins("<span class='adminnotice'>[key_name(usr)] [SSticker.delay_end ? "delayed the round end" : "has made the round end normally"].</span>")
+		var/reason = SSticker.admin_delay_notice //laziness
+		var/msg = "[key_name(usr)] delayed the round end.[reason ? " Reason: [reason]" : ""]"
+		log_admin(msg)

```

Comment:
Make this
`log_admin("[key_name(usr)] delayed the round end.[reason ? " Reason: [reason]" : ""]"`
`message_admins("[key_name_admin(usr)] delayed the round end.[reason ? " Reason: [reason]" : ""]"`